sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java

```diff
@@ -259,6 +261,17 @@ public static BlobItem populateBlobItem(BlobItemInternal blobItemInternal) {
         return blobItem;
     }
 
+    public static TaggedBlobItem populateTaggedBlobItem(FilterBlobItem filterBlobItem) {
+        Map<String, String> tags = new HashMap<>();
+        if (filterBlobItem.getTags() != null && filterBlobItem.getTags().getBlobTagSet() != null) {
+            for (BlobTag tag : filterBlobItem.getTags().getBlobTagSet()) {
```

Comment:
Could we consolidate this logic (BlobTagSet -> Map of tags) into another method in ModelHelper? There's some duplicate logic in BlobAsyncClient.getTagsWithResponse and ModelHelper.populateBlobItem,